counsel.el

```diff
@@ -4019,30 +3983,169 @@ TREEP is used to expand internal nodes."
+(defvar counsel-outline-settings
+  '((emacs-lisp-mode
+     :outline-regexp ";;[;*]+[\s\t]+"
+     :outline-level counsel-outline-level-emacs-lisp)
+    (org-mode
+     :outline-title counsel-outline-title-org
+     :action counsel-org-goto-action)
+    (markdown-mode
+     :outline-title counsel-outline-title-markdown)
+    (latex-mode
+     :outline-title counsel-outline-title-latex))
+  "An alist holding `counsel-outline' settings for particular
+major modes.
+
+Each entry is a pair \(MAJOR-MODE . PLIST\).  `counsel-outline'
+checks whether an entry exists for the current buffer's
+MAJOR-MODE and, if so, loads the settings specified by PLIST
+instead of the default settings.  The following settings are
+recognized:
+
+- `:outline-regexp' is a regexp to match the beggining of an
+  outline heading.  It is only checked at the start of a line and
+  so need not start with `^'.  The default is to use the value of
+  the variable `outline-regexp'.
+
+- `:outline-level' is a function of no args to compute the level of
+  an outline heading.  It is called with point at the beginning
+  of `outline-regexp' and with the match data reflicting
+  `outline-regexp'.  The default is to use the value of the
+  variable `outline-level'.
+
+- `:outline-title' is a function of no args to get the title of an
+  outline heading.  It is called with point at the end of
+  `outline-regexp' and with the match data reflicting
+  `outline-regexp'.  The default is to use the function
+  `counsel-outline-title'.
+
+- `:action' is a function of one arg, a marker corresponding to the
+  beginning of the selected outline heading, performing
+  `counsel-outline''s action to jump to this heading.  The
+  default is to use the function `counsel-outline-action'.")
+
+(defun counsel-outline-title ()
+  "Default function used by `counsel-outline' to get the title of
+the current outline heading. See `counsel-outline-settings'."
+  (buffer-substring (point) (line-end-position)))
+
+(defun counsel-outline-title-org ()
+  "Function used by `counsel-outline' to get the title of the
+current outline heading in org-mode buffers. See
+`counsel-outline-settings'."
+  (apply 'org-get-heading (counsel--org-get-heading-args)))
+
+(defun counsel-outline-title-markdown ()
+  "Function used by `counsel-outline' to get the title of the
+current outline heading in markdown-mode buffers. See
+`counsel-outline-title'."
+  ;; `outline-regexp' is set by `markdown-mode' to match both setext
+  ;; (underline) and atx (hash) headings (see
+  ;; `markdown-regex-header').
+  (or (match-string 1) ; setext heading title
+      (match-string 5))) ; atx heading title
+        
+(defun counsel-outline-title-latex ()
+  "Function used by `counsel-outline' to get the title of the
+current outline heading in latex-mode buffers. See
+`counsel-outline-settings'."
+  ;; `outline-regexp' is set by `latex-mode' (see
+  ;; `LaTeX-outline-regexp') to match section macros, in which case we
```

Comment:
Did you mean to refer to the built-in variable `latex-outline-regexp` from `lisp/textmodes/tex-mode.el`, or are you referring to the external function `LaTeX-outline-regexp` from the AUCTeX package?

Reply:
The function from AUCTeX. But actually it also work with the built-in mode, provided we use the variable `latex-section-alist` instead of `LaTeX-section-alist` when the latter is not bound. The `outline-level` function of the built-in mode is less sophisticated than that of AUCTeX (the former always assumes the first section level is `\part` whereas the latter adapts this do the document class), but it works. Just committed the simple change to support both variants.